tests/test_integration.py

```diff
@@ -0,0 +1,540 @@
+import os
+import pathlib
+import re
+import subprocess
+import sys
+from typing import Any
+from typing import Dict
+from typing import Generator
+from typing import Optional
+from typing import Sequence
+from typing import Union
+
+import pytest
+from riot.riot import _T_CompletedProcess
+from typing_extensions import Protocol
+
+
+_T_Path = Union[str, "os.PathLike[Any]"]
+
+
+def run(
+    args: Union[str, Sequence[str]], cwd: _T_Path, env: Optional[Dict[str, str]] = None
+) -> _T_CompletedProcess:
+    return subprocess.run(
+        args,
+        env=env,
+        encoding=sys.getdefaultencoding(),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        cwd=cwd,
+        shell=isinstance(args, str),
+    )
+
+
+class _T_TmpRun(Protocol):
+    def __call__(
+        self,
+        args: Union[str, Sequence[str]],
+        cwd: Optional[_T_Path] = None,
+        env: Optional[Dict[str, str]] = None,
+    ) -> _T_CompletedProcess:
+        ...
+
+
+@pytest.fixture
+def tmp_run(tmp_path: pathlib.Path) -> Generator[_T_TmpRun, None, None]:
+    """Run a command by default in tmp_path."""
+
+    def _run(
+        args: Union[str, Sequence[str]],
+        cwd: Optional[_T_Path] = None,
+        env: Optional[Dict[str, str]] = None,
+    ) -> _T_CompletedProcess:
+        if cwd is None:
+            cwd = tmp_path
+        return run(args, cwd, env)
+
+    yield _run
```

Comment:
I don't think this needs to be a generator, you can just do `return _run`

Generator fixtures are good for things that need setup/teardown

(Unless, tmp_path gets torn down on return? I would have assumed fixture teardown happens after each test finishes, not after the scope of where it is used closes?)

Reply:
🤷‍♂️ returning equivalent for our purposes but yielding is easier to change if later on we do need to setup/teardown. I don't feel strongly though.

Reply:
I don't feel strongly either.